tests/ast-alignment/fixtures-to-test.ts

```diff
@@ -22,6 +22,65 @@ interface CreateFixturePatternConfig {
 
 const fixturesDirPath = path.join(__dirname, '../fixtures');
 
+class FixturesTester {
+  protected fixtures: FixturePatternConfig[] = [];
+
+  constructor() {}
+
+  public addFixturePatternConfig(
+    fixturesSubPath: string,
+    config: CreateFixturePatternConfig = {}
+  ) {
+    if (!fs.existsSync(path.join(fixturesDirPath, fixturesSubPath))) {
+      throw new Error(
+        `Registered path '${path.join(
+          __dirname,
+          fixturesSubPath
+        )}' was not found`
+      );
+    }
+
+    const ignore = config.ignore || [];
+    const fileType = config.fileType || 'js';
+    const ignoreSourceType = config.ignoreSourceType || [];
+
+    /**
+     * The TypeScript compiler gives us the "externalModuleIndicator" to allow typescript-estree do dynamically detect the "sourceType".
+     * Babel has similar feature sourceType='unambiguous' but its not perfect, and in some specific cases we sill have to enforce it.
+     * Known issues:
+     * - https://github.com/babel/babel/issues/9213
+     */
+    if (ignoreSourceType.length) {
+      ignore.push(...ignoreSourceType);
+      for (const fixture of ignoreSourceType) {
+        this.fixtures.push({
+          // It needs to be the full path from within fixtures/ for the pattern
+          pattern: `${fixturesSubPath}/${fixture}.src.${config.fileType}`,
+          ignoreSourceType: true
+        });
+      }
+    }
+
+    this.fixtures.push({
+      pattern: `${fixturesSubPath}/!(${ignore.join('|')}).src.${fileType}`,
+      ignoreSourceType: false
+    });
+  }
+
+  public getFixtures(): Fixture[] {
+    return this.fixtures
+      .map(fixture =>
+        glob
+          .sync(`${fixturesDirPath}/${fixture.pattern}`, {})
+          .map(filename => ({
+            filename,
+            ignoreSourceType: fixture.ignoreSourceType
+          }))
+      )
+      .reduce((acc, x) => acc.concat(x), []);
```

Comment:
i was thinking about replacing this with `flatMap` but i will have to add `esnext` lib to tsconfig